logger.go

```diff
@@ -64,104 +61,121 @@ func (l *Logger) WithPrefix(prefix string) *Logger {
 	l.printf(warningLog, l.pfx(format), l.extend(args)...)
 }
 
-// WarningfWithDepth is equivalent to the global WarningfWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) WarningfWithDepth(extraDepth int, format string, args ...interface{}) {
-	l.printfWithDepth(warningLog, extraDepth, l.pfx(format), l.extend(args)...)
-}
-
 // Error is equivalent to the global Error function, with the addition of prefix and data content from this Logger.
 func (l *Logger) Error(args ...interface{}) {
 	l.print(errorLog, l.extendWithPfx(args)...)
 }
 
-// ErrorWithDepth is equivalent to the global ErrorWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) ErrorWithDepth(extraDepth int, args ...interface{}) {
-	l.printWithDepth(errorLog, extraDepth, l.extendWithPfx(args)...)
+// GetErrorEvent is equivalent to the global GetErrorEvent function, with the addition of prefix and data content from this Logger.
+func (l *Logger) GetErrorEvent(args ...interface{}) Event {
+	return l.getEvent(errorLog, l.extendWithPfx(args))
+}
+
+// ErrorIf is equivalent to the global ErrorIf function, with the addition of prefix and data content from this Logger.
+func (l *Logger) ErrorIf(err error, args ...interface{}) {
+	if err != nil {
+		if args != nil {
+			args = append(args, ": ")
+		}
+		args = append(args, err)
+		l.print(errorLog, l.extendWithPfx(args)...)
+	}
 }
 
 // Errorln is equivalent to the global Errorln function, with the addition of prefix and data content from this Logger.
 func (l *Logger) Errorln(args ...interface{}) {
 	l.println(errorLog, l.extendWithPfx(args)...)
 }
 
-// ErrorlnWithDepth is equivalent to the global ErrorlnWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) ErrorlnWithDepth(extraDepth int, args ...interface{}) {
-	l.printlnWithDepth(errorLog, extraDepth, l.extendWithPfx(args)...)
-}
-
 // Errorf is equivalent to the global Errorf function, with the addition of prefix and data content from this Logger.
 func (l *Logger) Errorf(format string, args ...interface{}) {
 	l.printf(errorLog, l.pfx(format), l.extend(args)...)
 }
 
-// ErrorfWithDepth is equivalent to the global ErrorfWithDepth function, with the addition of prefix and data content from this Logger.
-func (l *Logger) ErrorfWithDepth(extraDepth int, format string, args ...interface{}) {
-	l.printfWithDepth(errorLog, extraDepth, l.pfx(format), l.extend(args)...)
+// ErrorfIf is equivalent to the global ErrorfIf function, with the addition of prefix and data content from this Logger.
+func (l *Logger) ErrorfIf(err error, format string, args ...interface{}) {
+	if err != nil {
+		format += ": %v"
+		args = append(args, err)
+		l.printf(errorLog, l.pfx(format), l.extend(args)...)
+	}
+}
+
+// Fatal is equivalent to the global Fatal  function, with the addition of prefix and data content from this Logger.
```

Comment:
very minor (but I know you notice such things): there is a double-space between the function name and 'function' in the comment to all Fatal* methods.